Add media seeking during playback performance tests

jernoble opened the pull request:
#### 05586bf64678227b4ff8fa315779757c96eaa4cc
<pre>
Add media seeking during playback performance tests
<a href="https://bugs.webkit.org/show_bug.cgi?id=278845">https://bugs.webkit.org/show_bug.cgi?id=278845</a>
<a href="https://rdar.apple.com/134914099">rdar://134914099</a>

Reviewed by Eric Carlson.

Add tests that validate the speed of seek operations while video is playing and paused.

* PerformanceTests/Media/SeekLatencyWhilePaused.html: Added.
* PerformanceTests/Media/SeekLatencyWhilePlaying.html: Added.
* PerformanceTests/Media/test-with-audio.mp4: Added.

Canonical link: <a href="https://commits.webkit.org/282931@main">https://commits.webkit.org/282931@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5e6bc470f0e38e3e9073134aadde58929581c3bf

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/64591 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/43956 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/17186 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/68612 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/15197 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/51715 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/15476 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/51958 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/10484 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/67657 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/40667 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/55891 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/32582 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/37332 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/14070 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/59260 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/13599 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/70311 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/8536 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/13102 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/59281 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/8570 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/55979 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/59466 "Found 196 new API test failures: /WebKitGTK/TestAuthentication:/webkit/Authentication/authentication-no-credential, /WebKitGTK/TestWebKitFindController:/webkit/WebKitFindController/next, /WebKitGTK/TestSSL:/webkit/WebKitWebView/tls-errors-redirect-to-http, /WebKitGTK/TestCookieManager:/webkit/WebKitCookieManager/cookies-changed, /WebKitGTK/TestLoaderClient:/webkit/WebKitWebView/load-bytes, /WebKitGTK/TestWebsiteData:/webkit/WebKitWebsiteData/handle-corrupted-local-storage, /WebKitGTK/TestSSL:/webkit/WebKitWebView/web-socket-tls-errors, /WebKitGTK/TestLoaderClient:/webkit/WebKitWebView/load-alternate-html-for-local-page, /WebKitGTK/TestDOMNodeFilter:/webkit/WebKitDOMNodeFilter/node-iterator, /WebKitGTK/TestDOMNode:/webkit/WebKitDOMNode/hierarchy-navigation ... (failure)") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/14274 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/7054 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/729 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/39767 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/40845 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/42028 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/40589 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

eric-carlson commented on PerformanceTests/Media/SeekLatencyWhilePlaying.html:
```diff
@@ -0,0 +1,59 @@
+    runTest();
+});
+
+function waitFor(object, event) {
+    return new Promise(resolve => {
+        object.addEventListener(event, resolve, {once: true});
+    })
+}
+
+function sleepFor(timeout) {
+    return new Promise(resolve => setTimeout(resolve, timeout));
+}
```
Nit: It would be useful to have these in a separate .js file so they can be shared

jernoble committed:
Add media seeking during playback performance tests
https://bugs.webkit.org/show_bug.cgi?id=278845
rdar://134914099

Reviewed by Eric Carlson.

Add tests that validate the speed of seek operations while video is playing and paused.

* PerformanceTests/Media/SeekLatencyWhilePaused.html: Added.
* PerformanceTests/Media/SeekLatencyWhilePlaying.html: Added.
* PerformanceTests/Media/test-with-audio.mp4: Added.

Canonical link: https://commits.webkit.org/282931@main
